src/osd/PGLog.h

```diff
@@ -1302,6 +1305,7 @@ struct PGLog : DoutPrefixProvider {
       true,
       &log,
       missing,
+      NULL, // we take care of rollforwarding in peering state
```

Comment:
Where is that?

Reply:
https://github.com/ceph/ceph/blob/39fd806cd45e991deac49c8a938dd6381306facf/src/osd/PeeringState.cc#L4064

this is for appending new log entries

Reply:
https://github.com/ceph/ceph/blob/39fd806cd45e991deac49c8a938dd6381306facf/src/osd/PeeringState.cc#L4064

@athanatos this is for appending new log entries

Reply:
That code in PeeringState isn't part of peering -- it's merging in log entries during normal operations from submit_log_entries (log entries for marking unfound objects lost or log entries encoding write error codes for handling duplicate ops, etc).  That rollback behavior has an entirely different purpose -- if a replica receives a log entry for something past last_backfill, it invalidates the rollback information.  Thus, I think that comment is misleading and should probably just be removed.

Also, use nullptr in general over NULL.

Reply:
> That code in PeeringState isn't part of peering -- it's merging in log entries during normal operations from submit_log_entries (log entries for marking unfound objects lost or log entries encoding write error codes for handling duplicate ops, etc). That rollback behavior has an entirely different purpose -- if a replica receives a log entry for something past last_backfill, it invalidates the rollback information. Thus, I think that comment is misleading and should probably just be removed.
> 
> Also, use nullptr in general over NULL.

aah I see, agreed, updated.